### DISCUSSION TITLE
Failed to create user: Database error creating new user

### DISCUSSION THREAD

**Discussion by boilermanc** [supabase/supabase#20325] [Questions] [answered]
Trying to add a new user through the web interface I get the error: Failed to create user: Database error creating new user. What I have done:

I had a function called handle_new_users. I think that got installed maybe when I added a flow from Futterflow. It had errors:
![Capture1](https://github.com/supabase/supabase/assets/656498/c7ce24f1-e126-4992-b80d-4e905eab56cb)
I changed that to this and get no errors:
```
BEGIN
    INSERT INTO public.profiles (id, email)
    VALUES (NEW.id, NEW.email);
    RETURN NEW;
END;
```
I tried to add a user again and I it has the same error and this is what I see in the database log:
![Capture3](https://github.com/supabase/supabase/assets/656498/ec109955-751f-4991-a34d-2b70c09248c0)

I added rls and still get that error. I disabled row level and get that error. 

I then added permissions through the command line and I see this in the table:
![Capture4](https://github.com/supabase/supabase/assets/656498/6c4ec835-d6d9-4e7e-90a2-014355f642b2)

![Capture5](https://github.com/supabase/supabase/assets/656498/5e59a25d-12e7-42bc-b2e6-b9e710c2f93c)

Still, get permission issues. Not sure where to look now. 

Any other info you need, let me know.

thanks!

**Comment by GaryAustin1** [comment, marked as answer]
Your function has to be security definer type.  The auth user does not have access to the public schema.
https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques

**Reply by boilermanc** [reply]
Thanks so much @GaryAustin1 that fixed it. here is the code for reference that worked. I added the logging statement but it didn't work. It shows red in the supabase interface. Would be nice if it worked. But at least I can add a new user now!


BEGIN
    -- Insert a new profile row with the user's ID and email
    INSERT INTO public.profiles (id, email)
    VALUES (NEW.id, NEW.email);
    RETURN NEW;
EXCEPTION
    WHEN others THEN
        -- Consider logging the error and then proceed
        RAISE NOTICE 'Failed to insert into profiles: %', SQLERRM;
        RETURN NEW;
END;

**Reply by GaryAustin1** [reply]
Use RAISE LOG instead.

**Comment by boilermanc** [comment]
that worked!

**Comment by adamcdy** [comment]
Just worked for me too after hours of trying to fix! 
Thanks
